pkg/operation/botanist/component/etcd/etcd_copy_backups_task.go

```diff
@@ -0,0 +1,142 @@
+// limitations under the License.
+
+package etcd
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/extensions"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CopyInterface interface {
+	component.DeployWaiter
+	SetBackupConfig(druidv1alpha1.StoreSpec)
+	SetSourceBackupConfig(druidv1alpha1.StoreSpec)
+}
+
+type CopyBackupsTaskValues struct {
+	Name         string
+	Namespace    string
+	TargetStore  druidv1alpha1.StoreSpec
+	SourceStore  druidv1alpha1.StoreSpec
+	MaxBackups   *uint32
+	MaxBackupAge *uint32
+}
+
+type etcdCopyBackupsTask struct {
+	values       *CopyBackupsTaskValues
+	logger       logrus.FieldLogger
+	client       client.Client
+	waitInterval time.Duration
+	waitTimeout  time.Duration
+
+	etcdCopyBackupsTask *druidv1alpha1.EtcdCopyBackupsTask
+}
+
+func NewEtcdCopyBackupsTask(
+	logger logrus.FieldLogger,
+	client client.Client,
+	values *CopyBackupsTaskValues,
+	waitInterval time.Duration,
+	waitTimeout time.Duration) CopyInterface {
+
+	return &etcdCopyBackupsTask{
+		values,
+		logger,
+		client,
+		waitInterval,
+		waitTimeout,
+		&druidv1alpha1.EtcdCopyBackupsTask{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      values.Name,
+				Namespace: values.Namespace,
+			},
+		},
+	}
+}
+
+func (e *etcdCopyBackupsTask) Deploy(ctx context.Context) error {
+	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, e.client, e.etcdCopyBackupsTask, func() error {
+		metav1.SetMetaDataAnnotation(&e.etcdCopyBackupsTask.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile)
+		metav1.SetMetaDataAnnotation(&e.etcdCopyBackupsTask.ObjectMeta, v1beta1constants.GardenerTimestamp, TimeNow().UTC().String())
+		e.etcdCopyBackupsTask.Spec.MaxBackupAge = e.values.MaxBackupAge
+		e.etcdCopyBackupsTask.Spec.MaxBackups = e.values.MaxBackups
+		e.etcdCopyBackupsTask.Spec.SourceStore = e.values.SourceStore
+		e.etcdCopyBackupsTask.Spec.TargetStore = e.values.TargetStore
+		return nil
+	})
+
+	return err
+}
+
+func (e *etcdCopyBackupsTask) Wait(ctx context.Context) error {
+	if err := extensions.WaitUntilObjectReadyWithHealthFunction(
+		ctx,
+		e.client,
+		e.logger,
+		// TODO: export this in a method and add better error descriptions
+		func(obj client.Object) error {
+			etcdCopyBackupsTask, ok := obj.(*druidv1alpha1.EtcdCopyBackupsTask)
+			if !ok {
+				return errors.New("object is not *druidv1alpha1.EtcdCopyBackupsTask")
+			}
+			if etcdCopyBackupsTask.Status.State == druidv1alpha1.TaskStateSucceeded {
```

Comment:
This func should wait until the state is either suceeded or failed:
```suggestion
			if etcdCopyBackupsTask.Status.State == druidv1alpha1.TaskStateSucceeded || etcdCopyBackupsTask.Status.State == druidv1alpha1.TaskStateFailed {
```
Note that it should be adapted to use `Conditions`, instead of `State` with the latest `etcd-druid` changes.